docs/design/features/native-hosting.md

```diff
@@ -343,11 +409,55 @@ int component_entry_point_fn(void *arg, int32_t arg_size_in_bytes);
+
+In .NET 5 it is allowed to call this helper on host context which came from `hostfxr_initialize_for_dotnet_command_line` which will make all the application assemblies available in the default load context. In such case it is recommended to only use this helper for loading plugins external to the application. Using the helper to load assembly from the application itself will lead to duplicate copies of assemblies and duplicate types.
+
+In .NET 5 it is also allowed to call this helper on the first host context (by passing `nullptr` for the host context). If the origin of the first host context is unknown it can be the application context as noted above and thus only external plugins should be loaded via the helper to avoid confusion.
+
 It is allowed to call the returned runtime helper many times for different assemblies or different methods from the same assembly. It is not required to get the helper every time. The implementation of the helper will cache loaded assemblies, so requests to load the same assembly twice will load it only once and reuse it from that point onward. Ideally components should not take a dependency on this behavior, which means components should not have global state. Global state in components is typically just cause for problems. For example it may create ordering issues or unintended side effects and so on.
 
 The returned native function pointer to managed method has the lifetime of the process and can be used to call the method many times over. Currently there's no way to unload the managed component or otherwise free the native function pointer. Such support may come in future releases.
 
 
+### Calling managed function
+In .NET 5 the hosting layer adds a new functionality which allows just getting a native function pointer for already available managed method. This is implemented in a runtime helper `hdt_get_function_pointer`. Calling the `hostfxr_get_runtime_delegate(handle, hdt_get_function_pointer, &helper)` returns a function pointer to the runtime helper with this signature:
+```C
+int get_function_pointer_fn(
+    const char_t *type_name,
+    const char_t *method_name,
+    const char_t *delegate_type_name,
+    void         *load_context,
+    void         *reserved,
+    /*out*/ void **delegate)
+```
+
+Calling this function will find the specified type in the default load context, locate the required method on it and return a native function pointer to that method. The method's signature can be specified via the delegate type name.
+* `type_name` - Assembly qualified type name to find
+* `method_name` - Name of the method on the `type_name` to find. The method must be `static` and must match the signature of `delegate_type_name`.
+* `delegate_type_name` - Assembly qualified delegate type name for the method signature, or null. If this is null, the method signature is assumed to be:
```

Comment:
We should also permit the `UNMANAGEDCALLERSONLY_METHOD` which was added in .NET 5.0.